timemachine/cpp/src/kernels/k_neighborlist.cuh

```diff
@@ -223,9 +231,11 @@ void __global__ k_find_blocks_with_ixns(
     // Determine if the row block can be translated into a periodic box
     // to optimize distance calculations
     // https://github.com/proteneer/timemachine/issues/320
-    const bool single_periodic_box =
-        (0.5f * bx - row_bb_ext_x >= cutoff && 0.5f * by - row_bb_ext_y >= cutoff &&
-         0.5f * bz - row_bb_ext_z >= cutoff);
+    // Only possible when performing upper triangle, as otherwise indices may not be ordered
+    // and will cause the kerenl to hang
```

Comment:
It will prevent the optimization from triggering. Since we are going to be passing idxs within CUDA code, it seemed safer to not require the idxs to always be sorted. Will look at the benchmarks to see how much of an impact it has.